Utilities/macrecovery/macrecovery.py

```diff
@@ -204,12 +205,16 @@ def save_image(url, sess, filename='', directory=''):
         'Cookie': '='.join(['AssetToken', sess])
     }
 
+    if not os.path.exists(DOWNLOAD_DIR):
```

Comment:
This change is inadequate. `directory` is passed as an argument, and you always override it. Perhaps you should change the function prototype to `directory=DOWNLOAD_DIR` instead of `directory=''`, and then create `directory` if it does not exist here.

Reply:
Alright. I made another change in the latest commit, since `save_image` function gets called with a `directory` override.

I dropped such override, since by default `directory` value is `com.apple.recovery.boot` in the function prototype, despite I don't think that's the best way. Suggestions are welcome